cryptol-saw-core/src/Verifier/SAW/Cryptol.hs

```diff
@@ -1792,3 +1801,31 @@ importFirstOrderValue t0 v0 = V.runEval mempty (go t0 v0)
 
     _ -> panic "importFirstOrderValue"
                 ["Expected finite value of type:", show t, "but got", show v]
+
+-- | Generate functions to construct newtypes in the term environment.
+-- (I.e., make identity functions that take the record the newtype wraps.)
+genNewtypeConstructors :: SharedContext -> Map C.Name Newtype -> Env -> IO Env
+genNewtypeConstructors sc newtypes env0 =
+  foldM genConstr env0 newtypes
+  where
+    genConstr :: Env -> Newtype -> IO Env
+    genConstr env nt = do
+      constr <- importExpr sc env (newtypeConstr nt)
+      let env' = env { envE = Map.insert (ntName nt) (constr, 0) (envE env)
+                     , envC = Map.insert (ntName nt) (newtypeSchema nt) (envC env)
+                     }
+      return env'
+    newtypeConstr :: Newtype -> C.Expr
+    newtypeConstr nt = foldr tFn fn (C.ntParams nt)
+      where
+        -- FIXME? What about the `ntConstraints` for the Newtype...?
```

Comment:
I think we can safely ignore the `ntConstraints` here.  They arise from the "well formed" side conditions on types mentioned in the newtype fields.